app/models/source_manager.rb

```diff
@@ -0,0 +1,18 @@
+class SourceManager
+  class << self
+    PROTOCOLS = ['http', 'https']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.